Add CLI commands to get and set gateway's log level

gbregman opened the pull request:
Fixes #385

caroav reviewed (commented):
@gbregman  what is the change is state.py? is it related?

gbregman commented:
@caroav after I reverted the OMAP changes most of the changes in state.py were gone. What's left there is not directly linked to the log level change. Only, during debugging I ran into annoying cases where I got an exception in the grpc call which were silently swallowed so I couldn't find the problem. So, I added try clauses in these calls to display the exception details so in case of a problem we would know about it.

Also, I removed the redundant "import logging" which is not needed any more after we moved to our own class for the logger.

baum commented on control/cli.py:
```diff
@@ -394,9 +403,93 @@ def gw_version(self, args):
+        try:
+            req = pb2.set_gateway_log_level_req(log_level=log_level)
+        except ValueError as err:
+            errstr = str(err)
+            if errstr.startswith("unknown enum label "):
+                errstr = errstr.removeprefix("unknown enum label ")
+                self.cli.parser.error(f"invalid log level {errstr}")
+            else:
+                self.cli.parser.error(f"{err}")
```
Could it be be simplified? 
```suggestion
            self.cli.parser.error(f"invalid log level {log_level}, error {err}")
```

gbregman replied:
I prefer to have clearer messages in case of an error to saving a few lines in the code. But fine, I will change that.

gbregman committed:
Add CLI commands to get and set gateway's log level

Fixes #385

Signed-off-by: Gil Bregman <gbregman@il.ibm.com>

gbregman committed:
Merge branch 'ceph:devel' into devel
